week5_missing_number/lib/missing_number.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+MY_ARRAY = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10].freeze
+MAX_LENGHT = 9
+
+class NumberFinder
+  class InvalidinputError < StandardError; end
+
+  def initialize(arr)
+    @arr = arr
+  end
+
+  def call
+    input_validation
+    find_missing_number
+  rescue InvalidinputError => e
+     e.message
+  end
+
+  private
+
+  def find_missing_number
+    MY_ARRAY - @arr
+  end
+
+  def input_validation
+   raise InvalidinputError, "invalid input" unless (@arr.is_a? (Array)) && @arr.size == MAX_LENGHT && @arr.all?{|el| el.is_a? Integer}
```

Comment:
very long line. And if you have a long condition you can move it to a new method. For example:
```
def check?
  (@arr.is_a? (Array)) && @arr.size == MAX_LENGHT && @arr.all?{|el| el.is_a? Integer}
end

def input_validation
  raise InvalidinputError, "invalid input" unless check?
end
```

Reply:
I see, updated the code